### PR TITLE
Make libgit2 work on Windows Vista again

### PR DESCRIPTION
**Pull request by csware**
fixes issue #3316

### CODE REVIEW

**Comment by csware** [comment]
I've no clue why the OSx tests failed as the changes do not affect that platform.

File: src/path.c [hunk 1]
```diff
@@ -1099,6 +1100,8 @@ int git_path_diriter_init(
 	git_win32_path path_filter;
 	git_buf hack = {0};
 
+	int is_win7_or_later = git_has_win32_version(6, 1, 0);
```
Comment (vmg) [inline src/path.c]: Can you cache this?

Comment (csware) [inline src/path.c]: How? Just adding `static` doesn't work as git_has_win32_version is no constant.

Comment (vmg) [inline src/path.c]:
```
static int is_win7_or_later = -1;

if (is_win7_or_later < 0)
    is_win7_or_later = git_has_win32_version(6, 1, 0);
```

:)

**Comment by csware** [comment]
@vmg Caching added.

**Comment by csware** [comment]
The test cases only failed because of Network issues...

**Comment by csware** [comment]
I've no clue why the OSx tests failed again as the changes do not affect that platform.

**Comment by nulltoken** [comment]
Looks like an issue Travis side. Restarted the builds.

**Comment by csware** [comment]
The travis build failed again.
Are you going to merge this?

**Comment by whoisj** [comment]
I'm not sure about the description of this issue. While I fully support the fix, I'm not certain the issue is Vista vs Win7++. @csware have you tested the unfixed code on a non-VM installation of Vista? Does it reproduce?

**Comment by csware** [comment]
I don't have a real Vista here, but I don't see any difference to VM to real HW.

**Comment by ethomson** [comment]
Yep, I don't know why Travis is acting funny for Mac again.  Thanks for this!

**Comment by whoisj** [comment]
> IIRC the API description clearly states that those options are unsupported on Vista.

That is correct. However, I've spoken directly with the API team and those values should be ignored by older operating systems as part of the "down level driver support" that Vista and beyond support.

**Pull request merged**